barretenberg/cpp/src/barretenberg/dsl/acir_format/honk_recursion_constraint.hpp

```diff
@@ -7,55 +8,12 @@ using Builder = bb::UltraCircuitBuilder;
- * backend)
- * @param proof The honk proof. Is converted into circuit witness values (internal to the backend)
- * @param is_aggregation_object_nonzero A flag to tell us whether the circuit has already recursively verified proofs
- * (and therefore an aggregation object is present)
- * @param public_input The index of the single public input
- * @param input_aggregation_object Witness indices of pre-existing aggregation object (if it exists)
- * @param output_aggregation_object Witness indices of the aggregation object produced by recursive verification
- * @param nested_aggregation_object Public input indices of an aggregation object inside the proof.
- *
- * @note If input_aggregation_object witness indices are all zero, we interpret this to mean that the inner proof does
- * NOT contain a previously recursively verified proof
- * @note nested_aggregation_object is used for cases where the proof being verified contains an aggregation object in
- * its public inputs! If this is the case, we record the public input locations in `nested_aggregation_object`. If the
- * inner proof is of a circuit that does not have a nested aggregation object, these values are all zero.
- *
- * To outline the interaction between the input_aggergation_object and the nested_aggregation_object take the following
- * example: If we have a circuit that verifies 2 proofs A and B, the recursion constraint for B will have an
- * input_aggregation_object that points to the aggregation output produced by verifying A. If circuit B also verifies a
- * proof, in the above example the recursion constraint for verifying B will have a nested object that describes the
- * aggregation object in B’s public inputs as well as an input aggregation object that points to the object produced by
- * the previous recursion constraint in the circuit (the one that verifies A)
- *
- * TODO(https://github.com/AztecProtocol/barretenberg/issues/996): Update these comments for Honk.
- */
-struct HonkRecursionConstraint {
-    // In Honk, the proof starts with circuit_size, num_public_inputs, and pub_input_offset. We use this offset to keep
-    // track of where the public inputs start.
-    static constexpr size_t inner_public_input_offset = 3;
-    std::vector<uint32_t> key;
-    std::vector<uint32_t> proof;
-    std::vector<uint32_t> public_inputs;
-
-    friend bool operator==(HonkRecursionConstraint const& lhs, HonkRecursionConstraint const& rhs) = default;
-};
+// In Honk, the proof starts with circuit_size, num_public_inputs, and pub_input_offset. We use this offset to keep
+// track of where the public inputs start.
+static constexpr size_t HONK_RECURSION_PUBLIC_INPUT_OFFSET = 3;
 
 AggregationObjectIndices create_honk_recursion_constraints(Builder& builder,
-                                                           const HonkRecursionConstraint& input,
+                                                           const RecursionConstraint& input,
```

Comment:
maybe we should add an assert in this function that it's actually taking in a honk recursion constraint

Reply:
good idea. added